### ISSUE TITLE
Deprecation warning when connecting with 4.13.11

### ISSUE THREAD

**Issue by bgSosh** [Automattic/mongoose#6217]
**Do you want to request a *feature* or report a *bug*?**

Bug.

**What is the current behavior?**

I'm getting this message when I connect with latest release of v4:

```
`open()` is deprecated in mongoose >= 4.11.0, use `openUri()` instead, or set the `useMongoClient` option if using `connect()` or `createConnection()`. See http://mongoosejs.com/docs/connections.html#use-mongo-client
```

I found a previous report, but the issue was closed and the comments were locked.  

I checked the document link in the warning, but all I can see about this issue (under heading 'Option Changes in v5.x') relates to v5.  Please could you clarify what the situation is for latest v4?  Should I be adding useMongoClient in options?

**Please mention your node.js, mongoose and MongoDB version.**

Node 4.8.2, Mongoose 4.13.11, Mongodb 3.2,

**Comment by lineus** [comment]
Hi @bgSosh. 
to get around this issue you can add an options object as the 2nd arg to .connect() with useMongoClient: true.

here is an example:
`mongoose.connect('mongodb://localhost:27017/test', { useMongoClient: true })`

Thanks for pointing out the issue with the error message. I'll submit a pull request to update it to point to the 4.x docs [here](http://mongoosejs.com/docs/4.x/docs/connections.html#use-mongo-client).

**Comment by bgSosh** [comment]
Thanks @lineus - all clear now!

**Comment by bgSosh** [comment]
@lineus Hmm, ran into an issue with this - is there perhaps no createConnection that accepts both an option and a callback?

This works:
```
 dbConnection = mongoose.createConnection(connectionString_Forum, function(err) { 
       //Called back
 });
```
But this doesn't:
```
 dbConnection = mongoose.createConnection(connectionString_Forum, { useMongoClient: true }, function(err) { 
       //Not called back
 });
```

**Comment by lineus** [comment]
createConnection() doesn't take a callback in mongoose version 4. it only takes a uri and options and returns an instance of the native driver connection. Are you connecting to multiple dbs or is there some other compelling reason to use mongoose.createConnection() instead of mongoose.connect()?

**Comment by bgSosh** [comment]
Exactly, we need to connect to multiple dbs, and need a reference to the non-default connections.

Thanks for the clarification on createConnection().  Though, I can confirm that the callback is definitely called (when cb is the second parameter, but perhaps that's some unintended effect?  

Is there any reason why connect() and createConnection() should have these differences?

Could you tell the correct way to get an async notification of the connection event for createConnection in v4?

Thanks for your help.

**Comment by lineus** [comment]
it looks like when you call createConnection with a callback instead of options, it gets passed to connection.openUri, which checks if options is a function and reassigns it to the callback option. 
I'm probably not the best person to speak to the reasoning for historical decisions, so I'll leave that up to @vkarpov15 or @varunjayaraman. 

you can listen for the open event of the connection thusly:
```
const conn = mongoose.createConnection(uri, options)
conn.on('open', callback)
```

**Comment by bgSosh** [comment]
Thanks again @lineus.  Might be nice to have createConnection() match the supported signatures of connect() - maybe the devs you linked could comment on the feasibility of that.

**Comment by vkarpov15** [comment]
@lineus the fact that `useMongoClient` + `createConnection()` means you can't pass a callback is an unintentional bug. Will fix for 4.13.12 :+1:

**Comment by bgSosh** [comment]
Ah, great! thanks @vkarpov15, @lineus
